mobile/lib/services/preview_video_store.dart

```diff
@@ -0,0 +1,549 @@
+          );
+
+          FFProbeProps? props2;
+          props2 = await getVideoPropsAsync(playlistFile);
+          FileMagicService.instance.updatePublicMagicMetadata(
+            [enteFile],
+            {
+              previewHeightKey: props2?.height,
+              previewWidthKey: props2?.width,
+              previewSizeKey: objectSize,
+            },
+          ).ignore();
+          _logger.info("Video preview uploaded for $enteFile");
+        } catch (err, sT) {
+          error = "Failed to upload video preview\nError: $err";
+          _logger.shout("Video preview uploaded for $enteFile", err, sT);
+        }
+      } else if (ReturnCode.isCancel(returnCode)) {
+        _logger.warning("FFmpeg command cancelled");
+        error = "FFmpeg command cancelled";
+      } else {
+        final output = await session.getOutput();
+        _logger.shout(
+          "FFmpeg command failed with return code $returnCode",
+          output ?? "Error not found",
+        );
+        if (kDebugMode) {
+          _logger.severe(output);
+        }
+        error = "Failed to generate video preview\nError: $output";
+      }
+
+      if (error == null) {
+        _items[enteFile.uploadedFileID!] = PreviewItem(
+          status: PreviewItemStatus.uploaded,
+          file: enteFile,
+          retryCount: _items[enteFile.uploadedFileID!]!.retryCount,
+          collectionID: enteFile.collectionID ?? 0,
+        );
+      } else {
+        if (_items[enteFile.uploadedFileID!]!.retryCount < 3) {
+          _items[enteFile.uploadedFileID!] = PreviewItem(
+            status: PreviewItemStatus.retry,
+            file: enteFile,
+            retryCount: _items[enteFile.uploadedFileID!]!.retryCount + 1,
+            collectionID: enteFile.collectionID ?? 0,
+          );
+          files.add(enteFile);
+        } else {
+          _items[enteFile.uploadedFileID!] = PreviewItem(
+            status: PreviewItemStatus.failed,
+            file: enteFile,
+            retryCount: _items[enteFile.uploadedFileID!]!.retryCount,
+            collectionID: enteFile.collectionID ?? 0,
+            error: error,
+          );
+        }
+      }
+      Bus.instance.fire(PreviewUpdatedEvent(_items));
+    } finally {
+      if (uploadingFileId == enteFile.uploadedFileID!) {
+        uploadingFileId = -1;
+      }
+      if (files.isNotEmpty) {
+        final file = files.first;
+        files.remove(file);
+        await chunkAndUploadVideo(ctx, file);
+      }
+    }
+  }
+
+  Future<void> _reportVideoPreview(
+    EnteFile file,
+    File playlist, {
+    required String objectID,
+    required int objectSize,
+  }) async {
+    _logger.info("Pushing playlist for ${file.uploadedFileID}");
+    try {
+      final encryptionKey = getFileKey(file);
+      final playlistContent = playlist.readAsStringSync();
+      final result = await gzipAndEncryptJson(
+        {
+          "playlist": playlistContent,
+          'type': 'hls_video',
```

Comment:
The view preview height and size should be preserved here instead of file metadata as we might generate more previews of same file